### PR TITLE
SNOW-1445354: Add support for Series.str.match

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1445354

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Add support for Series.str.match.

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 1]
```diff
@@ -12239,8 +12239,51 @@ def str_fullmatch(
...
+        na : scalar, optional
+            Fill value for missing values. The default depends on dtype of the array. For object-dtype, numpy.nan is used. For StringDtype, pandas.NA is used.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler representing result of the string operation.
+        """
+        if not native_pd.isna(na) and not isinstance(na, bool):
+            ErrorMessage.not_implemented(
+                "Snowpark pandas doesn't support non-bool 'na' argument"
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:12263]: Can we specify the API?
```suggestion
                "Snowpark pandas doesn't support non-bool 'na' argument for .str.match()"
```

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM, just one suggestion

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py]: Good point. Done.

File: tests/integ/modin/series/test_str_accessor.py [hunk 2]
```diff
@@ -238,6 +238,44 @@ def test_str___getitem___string_key():
...
+    eval_snowpark_pandas_result(
+        snow_ser,
+        native_ser,
+        lambda ser: ser.str.match(pat, case=case, flags=flags, na=na),
+    )
+
+
+@pytest.mark.parametrize("na", [1, "klm", datetime.date(2019, 12, 4), [True]])
+@sql_count_checker(query_count=0)
+def test_str_match_invlaid_na(na):
```
Comment (sfc-gh-lmukhopadhyay) [inline tests/integ/modin/series/test_str_accessor.py:269]:
```suggestion
def test_str_match_invalid_na(na):
```
spelling nit

**Review by sfc-gh-lmukhopadhyay — APPROVED** [review]
just one nit but LGTM!

File: tests/integ/modin/series/test_str_accessor.py (hunk 2 above)
Comment (sfc-gh-nkrishna) [reply tests/integ/modin/series/test_str_accessor.py:269]: @sfc-gh-helmeleegy I think this got automerged, can you make this change in a follow-up PR (there are a few other string functions I think you're working on too)

**Pull request merged**